packages/tree-sitter-port/src/prepareNodeTypes.ts

```diff
@@ -0,0 +1,297 @@
+            const fieldNameB = multipleOrChildren[b][0]
+
+            const typesA = multipleOrChildren[a][1].nodeTypes
+            const typesB = multipleOrChildren[b][1].nodeTypes
+
+            const hasConflict = typesA.some((typeA) => typesB.includes(typeA))
+
+            if (hasConflict) {
+              let conflictingNodeType = null
+
+              for (let i = 0; i < typesA.length && !conflictingNodeType; i++) {
+                const typeAIndex = typesB.indexOf(typesA[i])
+
+                if (typeAIndex > -1) {
+                  conflictingNodeType = typesA[i]
+                }
+              }
+
+              if (
+                conflictingNodeType &&
+                !nodeTypesToIgnore.includes(conflictingNodeType)
+              ) {
+                /*
+                 * Conflict is when two or more fields with multiple children can have node of the same type within.
+                 * In this case we won't be able to properly assign nodes by type to given named field.
+                 * If conflict would occur for other parsers, we should put all node types from conflicting fields into common "children" field, as we cannot do anything better in this case.
+                 *
+                 * We don't care about conflicts between ignored node types, as they will not be processed anyway
+                 */
+                conflict = {
+                  nodeType: nodeType.type,
+                  fieldNameA,
+                  fieldNameB,
+                  conflictingNodeType,
+                  nodeTypesFilePath,
+                }
+              }
+            }
+          }
+        }
+      }
+
+      if (conflict !== null) {
+        const conflictObj = conflict // TS is freaking out
+
+        const conflictJsonString = JSON.stringify(conflictObj, null, 2)
+        const resolver = conflictResolvers.find(({ fields, nodeType }) => {
+          return (
+            fields.includes(conflictObj.fieldNameA) &&
+            fields.includes(conflictObj.fieldNameB) &&
+            nodeType === conflictObj.nodeType
+          )
+        })
+
+        if (resolver) {
+          const mergeSourceFieldName = [
+            conflictObj.fieldNameA,
+            conflict.fieldNameB,
+          ].filter((fieldName) => fieldName !== resolver.mergeToField)[0]
+
+          fieldsMeta[resolver.mergeToField].nodeTypes = uniq([
+            ...fieldsMeta[resolver.mergeToField].nodeTypes,
+            ...fieldsMeta[mergeSourceFieldName].nodeTypes,
+          ])
+
+          delete fieldsMeta[mergeSourceFieldName]
+
+          log('Resolved conflict ' + conflictJsonString)
+        } else {
+          throw new Error('Node types conflict: ' + conflictJsonString)
+        }
+      }
+    }
+
+    const data = {
+      named: nodeType.named,
+      type: nodeType.type,
+      fieldsMeta,
+    }
+
+    types[nodeType.type] = data
+  })
+
+  const uniqueNotNamedNodeTypesToInclude = new Set(notNamedNodeTypesToInclude)
```

Comment:
uniq(...) ??

Reply:
I used set, because we later search in this set. This is 'performance optimization' that actually does not matter for this script :D